pkg/cloud/cloud.go

```diff
@@ -957,218 +637,48 @@ func (c *cloud) getUpdateResizeVolumeAdministrativeAction(ctx context.Context, v
-							openZFSClientConfiguration.SetOptions(options)
-						default:
-							return nil, fmt.Errorf("%w: %s", errKeyDoesNotExist, key)
-						}
-					}
-
-					clientConfigurations = append(clientConfigurations, &openZFSClientConfiguration)
-				}
-
-				openZFSNfsExport.SetClientConfigurations(clientConfigurations)
-			default:
-				return nil, fmt.Errorf("%w: %s", errKeyDoesNotExist, clientKey)
-			}
-		}
-
-		nfsExports = append(nfsExports, &openZFSNfsExport)
-	}
-
-	return nfsExports, nil
+func CollapseCreateVolumeParameters(parameters map[string]string) (string, error) {
+	config := fsx.CreateOpenZFSVolumeConfiguration{}
+	from := util.ConvertStringMapToAny(parameters)
+	output, err := util.ConvertObjectToObjectToJsonString(from, &config)
+	return output, err
 }
 
-func (c *cloud) parseUserAndGroupQuotas(input string) ([]*fsx.OpenZFSUserOrGroupQuota, error) {
-	splitElements := util.SplitCommasAndRemoveOuterBrackets(input)
-
-	var userAndGroupQuotas []*fsx.OpenZFSUserOrGroupQuota
-
-	for _, element := range splitElements {
-		splitConfig := util.SplitCommasAndRemoveOuterBrackets(element)
-		mappedConfig, err := util.MapValues(splitConfig)
-		if err != nil {
-			return nil, fmt.Errorf("%w: %s", errInvalidMap, input)
-		}
-
-		openZFSUserOrGroupQuota := fsx.OpenZFSUserOrGroupQuota{}
-
-		for key, value := range mappedConfig {
-			if value == nil {
-				continue
-			}
-
-			switch key {
-			case "Id":
-				id, err := strconv.ParseInt(*value, 10, 64)
-				if err != nil {
-					return nil, fmt.Errorf("%w: %s=%s", errValueNotAnInt, key, *value)
-				}
-				openZFSUserOrGroupQuota.SetId(id)
-			case "StorageCapacityQuotaGiB":
-				storageCapacityQuotaGiB, err := strconv.ParseInt(*value, 10, 64)
-				if err != nil {
-					return nil, fmt.Errorf("%w: %s=%s", errValueNotAnInt, key, *value)
-				}
-				openZFSUserOrGroupQuota.SetStorageCapacityQuotaGiB(storageCapacityQuotaGiB)
-			case "Type":
-				openZFSUserOrGroupQuota.SetType(*value)
-			default:
-				return nil, fmt.Errorf("%w: %s", errKeyDoesNotExist, key)
-			}
-		}
-
-		userAndGroupQuotas = append(userAndGroupQuotas, &openZFSUserOrGroupQuota)
-	}
-
-	return userAndGroupQuotas, nil
+func CollapseDeleteVolumeParameters(parameters map[string]string) (string, error) {
+	config := fsx.DeleteVolumeOpenZFSConfiguration{}
+	from := util.ConvertStringMapToAny(parameters)
+	output, err := util.ConvertObjectToObjectToJsonString(from, &config)
+	return output, err
 }
 
-func (c *cloud) parseTags(input string) ([]*fsx.Tag, error) {
-	splitConfig := util.SplitCommasAndRemoveOuterBrackets(input)
-	mappedConfig, err := util.MapValues(splitConfig)
+func ValidateFileSystemDeleteParameters(parameters map[string]any) error {
+	input := fsx.DeleteFileSystemInput{}
+	err := util.ConvertObjectToObject(parameters, &input)
 	if err != nil {
-		return nil, fmt.Errorf("%w: %s", errInvalidMap, input)
+		return fmt.Errorf("%w. %s", ErrInvalidInput, err)
 	}
+	return input.Validate()
```

Comment:
Do we need this `input.Validate()`? I thought validation automatically occurred when we called the function which invokes the API?

Reply:
This is called during create. We want to validate the delete parameters manually before creating the object so the user can assume the delete parameters are correct.